csharp/src/Microsoft.ML.OnnxTraining/OrtValueCollection.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.ML.OnnxRuntime;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.ML.OnnxTraining
+{
+    /// <summary>
+    /// The OrtValueCollection holds ort values returned by the error function, but does not actually own any of them
```

Comment:
>ut does not actually own any of them [](start = 81, length = 36)

What is the reason for this class of being IDisposable?

Reply:
It does not need to be, however it does need the other SafeHandle support which by default implements the IDispose.

Reply:
I have now changed this to mirror the way you now implement the InferenceSession.cs with its _nativeHandle.